packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -180,9 +184,53 @@ function nodeRequire(
   root: string,
   preserveSymlinks: boolean
 ) {
-  const mod = require(resolve(id, importer, root, preserveSymlinks))
-  const defaultExport = mod.__esModule ? mod.default : mod
-  // rollup-style default import interop for cjs
+  const mdl = require(resolve(id, importer, root, preserveSymlinks));
+  return proxyESM(mdl);
+}
+
+function nodeRequireOrImport(
+  id: string,
+  importer: string | null,
+  root: string,
+  preserveSymlinks: boolean,
+  server: ViteDevServer,
+): Promise<any> {
+  return new Promise((res, reject) => {
+    if (id.startsWith('node:')) {
+      return res(NATIVE_IMPORT(id.slice('node:'.length)))
+    }
+    try {
+      const mdl = nodeRequire(id, importer, root, preserveSymlinks);
+      return res(mdl);
+    } catch (e) {
+      const resolvedPath = resolve(id, importer, root, preserveSymlinks);
+      if (e instanceof SyntaxError && /export|import/.test(e.message)) {
```

Comment:
https://github.com/vitejs/vite/pull/5197#issuecomment-934355948

Reply:
To be honest I'm on the fence about defaulting to `import`. I have updated the PR, and it does work, but I suspect there might be some compatibility issues lurking with this approach.

Internally, Node uses `cjs-module-lexer` to statically analyze CJS modules for named exports when using `import`. It's great, but obviously cannot be perfect. See [supported parsing examples](https://github.com/guybedford/cjs-module-lexer#parsing-examples).

The [Node docs](https://nodejs.org/api/esm.html#esm_import_statements) state (emphasis mine)...

> When importing CommonJS modules, the `module.exports` object is provided as the default export. Named exports **may** be available, provided by static analysis as a convenience for better ecosystem compatibility.